test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingFromQueryTest.cs

```diff
@@ -122,8 +122,12 @@ public class ModelBindingFromQueryTest
             // Assert
             var body = await response.Content.ReadAsStringAsync();
             var result = JsonConvert.DeserializeObject<Result>(body);
-            var error = Assert.Single(result.ModelStateErrors);
-            Assert.Equal("TestEmployees[0].EmployeeId", error);
+
+            Assert.Collection(
+                result.ModelStateErrors,
+                e => Assert.Equal("TestEmployees[0].EmployeeId", e),
+                e => Assert.Equal("TestEmployees[0].EmployeeTaxId", e),
+                e => Assert.Equal("TestEmployees[0].Age", e));
```

Comment:
This model is unfortunately used **all over**. It was simpler to just expect the errors you'd get from this set of inputs. This test was really *about* error messages anyway.